src/vs/platform/instantiation/common/instantiation.ts

```diff
@@ -132,6 +132,8 @@ export interface IInstantiationService {
 	createInstance<A1, A2, A3, A4, A5, A6, A7, T>(descriptor: descriptors.SyncDescriptor7<A1, A2, A3, A4, A5, A6, A7, T>, a1: A1, a2: A2, a3: A3, a4: A4, a5: A5, a6: A6, a7: A7): T;
 	createInstance<A1, A2, A3, A4, A5, A6, A7, A8, T>(descriptor: descriptors.SyncDescriptor8<A1, A2, A3, A4, A5, A6, A7, A8, T>, a1: A1, a2: A2, a3: A3, a4: A4, a5: A5, a6: A6, a7: A7, a8: A8): T;
 
+	createInstance<T>(descriptor: descriptors.SyncDescriptor0<T>, ...args: any[]): T;
```

Comment:
:-1: I think the evil lies in the fact that the actual logic is hidden in actions and not in a command-function. I would not restructure towards more actions but towards one command that does more or less this

```
// Handle in browser process
		if (this.ipcArg) {
			ipc.send(this.ipcMsg, this.ipcArg);
		} else {
			ipc.send(this.ipcMsg);
		}
```
and make the action use that command as well as the API using it. The relations should be `command_id -> function` and `action -> command_id` and `api_command -> command_id`. We have done with language features similar things (see suggest for instance)

Maybe we should talk about this ;-)